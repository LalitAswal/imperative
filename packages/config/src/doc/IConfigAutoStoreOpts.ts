/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { ICommandArguments } from "../../../cmd/src/doc/args/ICommandArguments";
import { IHandlerParameters } from "../../../cmd/src/doc/handler/IHandlerParameters";

export interface IConfigAutoStoreFindActiveProfileOpts {
    params?: IHandlerParameters;
    profileProps?: string[];
    profileTypes?: string[];
    defaultProfileName?: string;
}

export interface IConfigAutoStoreFindAuthHandlerForProfileOpts extends IConfigAutoStoreFindActiveProfileOpts {
    profilePath?: string;
    cmdArguments?: ICommandArguments;
    defaultBaseProfileName?: string;
}

export interface IConfigAutoStoreStoreSessCfgPropsOpts extends IConfigAutoStoreFindAuthHandlerForProfileOpts {
    sessCfg?: { [key: string]: any };
    propsToStore?: string[];
    profileName?: string;
    profileType?: string;
}

