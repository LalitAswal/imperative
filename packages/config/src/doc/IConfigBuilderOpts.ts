/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { IConfig } from "./IConfig";
import { IProfileProperty } from "../../../profiles";

export interface IConfigBuilderOpts {
    /**
     * Specify true to populate default values for profile properties with the
     * `IProfileProperty.includeInTemplate` flag set to true.
     */
    populateProperties?: boolean;

    /**
     * Existing config properties that the new config will be merged with. If
     * set, the config builder skips prompting for properties that already have
     * values defined.
     */
    mergeConfig?: IConfig;

    /**
     * Callback that prompts the user to enter a value for a profile property.
     * The method will be called to populate missing values in the base profile.
     * It should return the value that the user entered, or null or undefined if the prompt was cancelled.
     * @param propName The name of the property
     * @param property The profile property definition
     */
    getValueBack? (propName: string, property: IProfileProperty): Promise<any>;
}
