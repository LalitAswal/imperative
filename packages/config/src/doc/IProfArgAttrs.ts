/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { IProfLoc } from "./IProfLoc";

/**
 * The attributes of a profile argument.
 */
export interface IProfArgAttrs {
    /** The name of the argument */
    argName: string;

    /** The type of data for this property */
    dataType: "string" | "number" | "boolean" | "object";

    /** The value for the argument */
    argValue: string | number | boolean | object;

    /** The location of this argument */
    argLoc: IProfLoc;

    /** Whether the argument value is stored securely */
    isSecure?: boolean;
}
