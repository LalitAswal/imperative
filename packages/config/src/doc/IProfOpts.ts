/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

/**
 * Options that will affect the behavior of the ProfileInfo class.
 * They are supplied on the ProfileInfo constructor.
 */
export interface IProfOpts {
    /**
     * Indicates if environment variables should override values
     * stored in Zowe configuration profiles on disk.
     * The default is false.
     */
    overrideWithEnv?: boolean;
}
