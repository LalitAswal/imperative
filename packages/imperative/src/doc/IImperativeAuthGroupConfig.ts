/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { ICommandDefinition } from "../../../cmd";

export interface IImperativeAuthGroupConfig {
    /**
     * Command definition for "auth" command group
     */
    authGroup?: Partial<ICommandDefinition>;

    /**
     * Command definition for "auth login" command group
     */
    loginGroup?: Partial<ICommandDefinition>;

    /**
     * Command definition for "auth logout" command group
     */
    logoutGroup?: Partial<ICommandDefinition>;
}
