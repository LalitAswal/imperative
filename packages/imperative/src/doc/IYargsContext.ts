import * as net from "net";


/**
 * Allow for passing our own "context" / user data through yargs
 * @export
 * @interface IYargsContext
 */
export interface IYargsContext {

    /**
     * Stream to write response to
     * @type {net.Socket}
     * @memberof IYargsContext
     */
    stream?: net.Socket;

    /**
     * Current working directory from socket client
     * @type {string}
     * @memberof IYargsContext
     */
    cwd?: string;
};