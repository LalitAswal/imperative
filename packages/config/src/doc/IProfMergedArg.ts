/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { IProfArgAttrs } from "./IProfArgAttrs";

/**
 * The result object from mergeProfileArgs().
 */
export interface IProfMergedArg {
    /**
     * The list of arguments with known values.
     * All of the attributes in IProfPropAttrs will be filled in except
     * when knownArgs[i].propLoc.osLoc (and/or jsonLoc) are not
     * relevant for the type of location (locType).
     */
    knownArgs: IProfArgAttrs[];

    /**
     * The list of required arguments for which no value has been specified.
     * Obviously, the missingArgs[i].propValue attribute will not exist.
     * Note that a generated team configuration template can contain some
     * arguments with an empty string as a value. Such arguments will be
     * contained in this missing list. The other missing arguments will
     * have a missingArgs[i].propLoc, derived from the location of the
     * profile specified to the function mergeProfileArgs().
     */
    missingArgs: IProfArgAttrs[];

    /**
     * The list of arguments with secure values.
     * All of the attributes in IProfPropAttrs will be filled in where possible,
     * except for argValue since the values are sensitive data. Use the method
     * ProfileInfo.credentials.loadSecure to load the value of a secure arg.
     */
    secureArgs: IProfArgAttrs[];
}
