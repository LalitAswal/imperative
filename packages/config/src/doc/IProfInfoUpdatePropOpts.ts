/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { IProfArgValue } from "./IProfArgAttrs";
import { IProfMergedArg } from "./IProfMergedArg";

/**
 * Required options to update any property from the ProfileInfo class.
 */
export interface IProfInfoUpdatePropOpts  {
    /**
     * Property to update
     */
    property: string;

    /**
     * Value to use when updating the given property
     */
    value: IProfArgValue;

    /**
     * Type of the active profile
     */
    profileType: string;

    /**
     * Name of the active profile
     */
    profileName: string;

    /**
     * Specifies if the property should be stored securely or not
     */
    setSecure?: boolean;
}

/**
 * Required options to update known properties from the ProfileInfo class.
 */
export interface IProfInfoUpdateKnownPropOpts  {
    /**
     * Merged arguments object describing the location of the property to update
     */
    mergedArgs: IProfMergedArg;

    /**
     * Property to update
     */
    property: string;

    /**
     * Value to use when updating the given property
     */
    value: IProfArgValue;

    /**
     * Specifies if the property should be stored securely or not
     */
    setSecure?: boolean;
}
