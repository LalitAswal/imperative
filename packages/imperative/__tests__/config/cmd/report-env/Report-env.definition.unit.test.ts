/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { ICommandDefinition } from "../../../../../cmd/src/doc/ICommandDefinition";

describe("report-env command definition", () => {
    it("should have the right command content", () => {
        const numOfDaemonCmds = 0;
        const definition: ICommandDefinition = require("../../../../src/config/cmd//report-env/report-env.definition");
        expect(definition).toBeDefined();
        expect(definition).toMatchSnapshot();
    });
});
