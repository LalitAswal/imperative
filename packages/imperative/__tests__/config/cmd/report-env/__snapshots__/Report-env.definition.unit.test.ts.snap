// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`daemon group definition should have the right command content 1`] = `
Object {
  "ReportEnvDefinition": Object {
    "aliases": Array [
      "re",
    ],
    "description": "Reports key items from your environment and identifies problem conditions",
    "examples": Array [
      Object {
        "description": "Report information and issues about your working environment",
        "options": "",
      },
    ],
    "handler": "C:\\\\ourstuff\\\\repos\\\\imperative\\\\packages\\\\imperative\\\\src\\\\config\\\\cmd\\\\report-env\\\\report-env.handler",
    "name": "report-env",
    "summary": "Report the state of your working environment",
    "type": "command",
  },
}
`;
